app.js

```diff
@@ -41,35 +41,55 @@ addMushroomButton.addEventListener('click', () => {
 });
 
 addFriendButton.addEventListener('click', () => {
-    // get the name from the input
-    // create a new friend object
-    // push it into the friends state array, passed in as an argument
-    // reset the input
-    // display all the friends (use a function here)
+    
+    let friendName = friendInputEl.value;
+    
+    const friend = {
+        name: friendName,
+        satisfaction: 1
+    };
+
+    if (friendName === '') {
+        friend.name = `Friend ${Math.floor(Math.random() * 1000)}`;
+    }
+    
+    friendData.push(friend);
+   
+    
+    friendName = '';
+    
+    displayFriends();
 });
 
 function displayFriends() {
-    // clear out the friends in DOM
-
-    // for each friend in state . . .
+    
+    friendsEl.textContent = '';
+    
     for (let friend of friendData) {
-        // use renderFriend to make a friendEl
+        
+        const renderedFriend = renderFriend(friend);
+        
+        renderedFriend.addEventListener('click', () => {
+            if (friend.satisfaction < 3 && mushroomCount > 0) {
+                friend.satisfaction++;
+                mushroomCount--;
+            }
```

Comment:
Hi Kyle, you are just missing conditions (and accompanying alerts) here for if a friend is satisfied and can not eat anymore mushrooms, and if there are no more mushrooms and you need to forage for more.